src/signals/my-incidents/components/EmailInput.tsx

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2022 Gemeente Amsterdam
+import { useRef, useState } from 'react'
+
+import { useHistory } from 'react-router-dom'
+import * as yup from 'yup'
+
+import Button from 'components/Button'
+import Label from 'components/Label'
+
+import { useMyIncidentContext } from '../context'
+import { routes } from '../definitions'
+import { usePostEmail } from '../hooks'
+import { ErrorWrapper, StyledErrorMessage, StyledInput } from './styled'
+import type { Error } from './types'
+
+const schema = yup.string().email().required().max(254)
+
+const validateInput = async (
+  inputRef: React.RefObject<HTMLInputElement>,
+  setValidation: (error: Error) => void
+) => {
+  const isValid = await schema
+    .isValid(inputRef?.current?.value)
+    .then((valid) => valid)
+
+  if (isValid) {
+    setValidation({
+      hasError: !isValid,
+      message: null,
+    })
+  } else {
+    setValidation({
+      hasError: !isValid,
+      message: 'Het veld moet een geldig e-mailadres bevatten',
+    })
+  }
+}
+
+export const EmailInput = () => {
```

Comment:
Actually think using react hook form would be a good fit. Reduces a lot of complexity and aligns more with the logic of the other form components.